### PR TITLE
Fix vanishing second derivative of tanh and gradient of i0 for inputs

### LINKED ISSUE #39801: jax second derivative of tanh returns zero for tiny float64 input
### Description

`jax.numpy.tanh` appears to return an incorrect second derivative for a tiny positive `float64` input.

The second derivative of `tanh(x)` is:

```
tanh''(x) = -2 * tanh(x) * (1 - tanh(x)^2)
```

Near zero, this is approximately:

```
tanh''(x) ~= -2x
```

For `x = 1e-300`, the expected second derivative is approximately `-2e-300`, which is finite and representable in `float64`. However, JAX returns `0.0`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

try:
    import jax
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

x = 1e-300
expected = -2e-300

d1 = jax.grad(lambda z: jnp.tanh(z))
d2 = jax.grad(d1)

out = float(d2(jnp.asarray(x, dtype=jnp.float64)))

print("jax:", out)
print("expected:", expected)

if not (math.isfinite(out) and abs(out - expected) <= 1e-6 * abs(expected)):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
jax: 0.0
expected: -2e-300
```

### Expected result

```
-2e-300
```

The expected value is finite and representable in `float64`. JAX instead returns `0.0`, suggesting that the higher-order autodiff path loses the tiny but nonzero second derivative.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

### PR DESCRIPTION
**Pull request by lihe07**
Fixes #39801 and #39797. These two JVP rules calculate the derivative using ans instead of x for performance reasons. But the resulting expressions are sensitive to rounding errors.
Without adding expensive ops, this PR simply rewrites two expressions to make them correct under sensitive x.

### tanh second derivative 39801

The default JVP was `(1 + ans) * (1 - ans)`, but the autodiff for a second derivative produces `ans' * ((1 - ans) - (1 + ans))`. For small `-eps < x < eps`,  1 +- ans rounds to 0, so the true `-2 ans` cancels to zero. 

```py
jax.grad(jax.grad(jnp.tanh))(1e-300)
0.0        # expected -2e-300
```

Rewriting the expr to equivalently `1 - ans * ans` preserves the derivative.

```py
jax.grad(jax.grad(jnp.tanh))(1e-300)
-2e-300
```

### i0 gradient 39797

_i0 is computed as exp(|x|) * i0e(|x|), and the derivative was i1e(x) - sign(x) * i0e(x), which rounds to exactly -1.0 for |x| << 1. The product rule then yields 1*1 + 1*(-1) = 0.

```py
jax.grad(jnp.i0)(1e-300)
0.0        # expected 5e-301
```

Stating the identity d/dx i0 = i1 avoids the composition

```py
jax.grad(jnp.i0)(1e-300)
5e-301
```

### Testing

Adds testTanhSecondGrad to tests/lax_test.py.

### CODE REVIEW

**Comment by hawkinsp** [comment]
The problem with this approach is that it makes the first derivative less precise as ans -> 1. And that's not a good trade to make (making the first derivative of a very common operation worse in service of making the second derivative more precise).

I have a thought on how to fix this; I will send a PR for it soon.
